### ISSUE TITLE
[bug] Conan installation fails with python 3.13 and pip

### ISSUE THREAD

**Issue by Utosz** [conan-io/conan#17220]
### Describe the bug

Windows 10 Enterprise , Python 3.13.0, pip 24.2

Hi all,

i would like to switch from Python 3.8 to the newest Python 3.13 as the first one lost its support.
But before it i wanted to do a research if all of my subdependencies are working with it. Unfortunately i have a problem with Conan installation via pip. In my case im trying to use latest conan<2.0 but it also doesnt work for newest version.

pip install "conan<2.0" Output:
```
Collecting conan<2.0
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [14 lines of output]
      error: Multiple top-level packages discovered in a flat-layout: ['ccm', 'CSC', 'IME', 'INF', 'OCR', 'PLA', 'SKB', 'Vss', 'Web', 'Boot', 'Help', 'Logs', 'TAPI', 'Temp', 'WaaS', 'debug', 'Fonts', 'Media', 'Setup', 'Tasks', 'addins', 'Speech', 'System', 'WinSxS', 'CbsTemp', 'Cursors', 'Panther', 'schemas', 'tracing', 'wlansvc', 'apppatch', 'assembly', 'bcastdvr', 'Branding', 'ELAMBKUP', 'Firmware', 'IMECache', 'Minidump', 'Prefetch', 'rescache', 'SchCache', 'security', 'System32', 'SysWOW64', 'twain_32', 'appcompat', 'DiagTrack', 'InboxApps', 'Installer', 'L2Schemas', 'Migration', 'ModemLogs', 'Resources', 'servicing', 'Containers', 'SystemApps', 'SystemTemp', 'diagnostics', 'IdentityCRL', 'InputMethod', 'Performance', 'PrintDialog', 'AppReadiness', 'Provisioning', 'Registration', 'ServiceState', 'DigitalLocker', 'Globalization', 'RemotePackages', 'Speech_OneCore', 'SensorFramework', 'ServiceProfiles', 'ShellComponents', 'SystemResources', 'ShellExperiences', 'LiveKernelReports', 'PolicyDefinitions', 'LanguageOverlayCache', 'SoftwareDistribution', 'GameBarPresenceWriter', 'ImmersiveControlPanel', 'BitLockerDiscoveryVolumeContents'].

      To avoid accidental inclusion of unwanted files or directories,
      setuptools will not proceed with this build.

      If you are trying to create a single distribution with multiple packages
      on purpose, you should not rely on automatic discovery.
      Instead, consider the following options:

      1. set up custom discovery (`find` directive with `include` or `exclude`)
      2. use a `src-layout`
      3. explicitly set `py_modules` or `packages` with a list of names

      To find more information, look for "package discovery" on setuptools docs.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

### How to reproduce it

Download & Install Python 3.13 from https://www.python.org/downloads/.
choose python 3.13 and create venv:
`python -m venv .venv`
 activate venv:
`.venv\Scripts\activate`
install conan<2.0:
`pip install "conan<2.0"`
or install newest available in pypi:
`pip install conan`

**Comment by memsharded** [comment]
Hi @Utosz 

Thanks for your feedback.

There are no plans to support Conan 1.X with the very latest Python 3.13, this will be quite challenging due to some dependencies, which cannot easily be upgraded without breaking behavior. The recommendation is to upgrade to Conan 2, it is already in the Conan 2.9 release after more than 20 months and dozens of releases since, and it is the recommended version for production.

If you have any issues with Python 3.13 for Conan 2.X, please let us know. Thanks!

**Comment by Utosz** [comment]
Hey thanks for your feedback @memsharded !
Unfortunately i have the same problem with conan>2 as stated in description:
```In my case im trying to use latest conan<2.0 but it also doesnt work for newest version.```

**Comment by memsharded** [comment]
I am trying to reproduce. I just installed Python 3.13 in a venv in Windows, it comes with pip 22.3.1 (not 24.2, did you installed it manually or something?).

Then my ``pip`` fails way before your point, and it is not even related to Conan, it fails to ``pip install`` anything.

**Comment by memsharded** [comment]
After updating pip version, I get a similar error (not the same, but similar).
But this seems it is not even using any conan files yet, just failing in the installation of dependencies, it seems the error could be in the ``patch-ng`` dependency. 
The error can be reproduced with:
```
$ python -m pip install "patch-ng<1.19,>=1.18.0"
```

We maintain a fork of this, so we might be able to fix this, I'll have a look

**Comment by memsharded** [comment]
Re-opening until the new version of python-patch-ng is out there and we can confirm this is solved

**Comment by uilianries** [comment]
@Utosz Hello! We just released a new patch version for patch-ng fixing your issue:

https://github.com/conan-io/python-patch-ng/releases/tag/1.18.1

Could you please update your local copy, then, install Conan 2.x as usual:

    pip install -U patch-ng
    pip install -U conan

**Comment by memsharded** [comment]
For some reason now I manage do make it work with the ``patch-ng==1.18.0`` version.
Are you sure it has been published to PyPI with the new 1.18.1 version?

**Comment by memsharded** [comment]
Yes, it seems everything is good with the 1.18.1 version in PyPI.
I have no idea why it worked now with 1.18.0

**Comment by uilianries** [comment]
@memsharded  I was not able to reproduce the case in Linux, using same python and pip versions with patch-ng 1.18.0. However, @czoido commented he was able to reproduce it on Windows, but not in Mac. It seems be related to Windows only.

**Comment by czoido** [comment]
> @memsharded I was not able to reproduce the case in Linux, using same python and pip versions with patch-ng 1.18.0. However, @czoido commented he was able to reproduce it on Windows, but not in Mac. It seems be related to Windows only.

In fact, I couldn't reproduce it on any platform, but let's hope this fix resolves the issue!

**Comment by czoido** [comment]
Hi @Utosz,

Could you please check if this solves the problem after the update in `patch-ng` so we can close the issue?
Doing:

```
pip install -U patch-ng
pip install -U conan
```

Thank you so much!

**Comment by memsharded** [comment]
Hi @Utosz 

Can you please check if the issue is gone now? Thanks for your feedback.

**Comment by Utosz** [comment]
Hey,
thanks for your feedback!
I checked it and now the same problem with installing patch-ng:
![image](https://github.com/user-attachments/assets/8b139a6f-9b1c-4665-af21-155a62d53545)
![image](https://github.com/user-attachments/assets/a408504f-3740-45e9-a256-ef1e429b0205)
![image](https://github.com/user-attachments/assets/d0813bed-92df-48c5-9c38-65da52ddfc13)

**Comment by memsharded** [comment]
I don't know what is happening, I am trying to reproduce it here, but everything seems to work fine, I cannot make it fail.

Are you sure your virtualenv is correctly set-up? Can you please try:
- Make sure to purge your cache: ``python -m pip cache purge``
- Make sure to use the pip belonging to the current environment: ``python -m pip install patch-ng``

**Comment by dimbleby** [comment]
The preferred way to avoid any possibility of this sort of thing happening would be for the project to publish wheels itself.  Then there can be no possible way for that to go wrong on user machines.

**Comment by memsharded** [comment]
I am rechecking this. 
Lets try to have full updated reproducible steps. 
This is what I am trying:
```bash
# create venv
$ C:\ws\Python\Python313\python -m venv ./conantest313

# activate venv
$ conantest313\Scripts\activate.bat

# check versions
$ python --version
Python 3.13.0
$ pip --version
pip 24.2 from C:\Users\memsharded\conanws\conantest313\Lib\site-packages\pip (python 3.13)

$ conan --version
'conan' is not recognized as an internal or external command, operable program or batch file.

$ pip install conan
Collecting conan
  Using cached conan-2.11.0.tar.gz (476 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting requests<3.0.0,>=2.25 (from conan)
  ...
Building wheels for collected packages: conan
  Building wheel for conan (pyproject.toml) ... done
  Created wheel for conan: filename=conan-2.11.0-py3-none-any.whl size=605029 sha256=d3efd3cf051f845d2d1e3689601430cee8b060e9c9af8a32e8edb14412cf0180

Successfully installed... conan-2.11.0 ...
```

Can you please @Utosz do these exact steps in your machine? Thanks!

**Comment by memsharded** [comment]
I am closing this ticket as not reproducible.

Last comment https://github.com/conan-io/conan/issues/17220#issuecomment-2604884401 gave some exact steps to try to reproduce.

If someone can give similar steps so we can reproduce from our side, please share them here and I will re-open.

Thanks very much for the feedback.

**Comment by dimbleby** [comment]
You still should publish wheels
